src/kw/kw.cpp

```diff
@@ -173,3 +240,77 @@ string KanoWorld::whoami(void)
 {
     return string("");
 }
+
+
+/**
+ *   Collects and returns the token from the Server Response data
+ */
+string KanoWorld::get_token(void)
+{
+    JSON_Value *schema = json_parse_string(server_response.c_str());
```

Comment:
Smart pointer here to prevent memory leaks:

```suggestion
    std::shared_ptr<JSON_Value> schema(
        json_parse_string(server_response.c_str()),
        json_value_free);
```